src/pages/api/auth-connect/mark-identity-as-primary.ts

```diff
@@ -42,27 +56,37 @@ const handler: Handler = async (request, response) => {
           true,
         );
 
-        return markIdentityAsPrimary(
-          config.managementCredentials,
-          identityId,
-        ).then(() => {
-          span.setDisclosedAttribute("is Identity marked as primary", true);
+        return markIdentityAsPrimary(config.managementCredentials, identityId)
+          .then(() => {
+            span.setDisclosedAttribute("is Identity marked as primary", true);
 
-          response.statusCode = HttpStatus.OK;
-          response.setHeader("Content-type", "application/json");
-          response.end();
-          return;
-        });
+            response.statusCode = HttpStatus.OK;
+            response.setHeader("Content-type", "application/json");
+            response.end();
+            return;
+          })
+          .catch((error) => {
+            if (error instanceof GraphqlErrors) {
+              span.setDisclosedAttribute("Identity not found", error.message);
+              throw webErrorFactory(webErrors.identityNotFound);
+            }
+
+            if (error instanceof ConnectUnreachableError) {
+              span.setDisclosedAttribute("Connect unreachable", error.message);
+              throw webErrorFactory(webErrors.connectUnreachable);
+            }
+
+            span.setDisclosedAttribute("is Identity marked as primary", false);
+            throw error;
```

Comment:
I have questions:
- should the `span.setDisclosedAttribute("is Identity marked as primary", false);` be at the top if we want it to be on traces even if we throw an error?
- shouldn't we have the same naming for an attribute that is an error message? to facilitate looking for it? 

The recovery middleware uses:
```

          span.setDisclosedAttribute("exception.class", error.toString());
          span.setDisclosedAttribute("exception.message", error.message);
          span.setDisclosedAttribute("stack_trace_hash", error.stack);
          ```

Reply:
For the 1st point, I totally agree, these kind of spans should be on top.

And for the second one, it might be useful indeed if we can filter by span names. So we go for `exception.message` when we send the error message ?

Reply:
We could create a function that takes the span, the error and add the attributes to the span.

Reply:
> We could create a function that takes the span, the error and add the attributes to the span.

Well the error middleware does that, but it doesn't unwrap the error that made us use the weberror (maybe something to deal with in the future)

Reply:
What do you mean by `unwrap`?

Reply:
```
if (error instanceof ConnectUnreachableError) {
  span.setDisclosedAttribute("Connect unreachable", error.message);
  throw webErrorFactory(webErrors.connectUnreachable);
}
```
in this example, the error middleware will only see the webError, it won't have access to the `ConnectUnreachableError` . You could do something like `throw webErrorFactory(webErrors.connectUnreachable, error)` and create a webError that uses the `parentError` option (I think we have one). But I don't think we use this in the error middleware so we display the WebError but not what has caused the WebError. So we would need to 'unwrap' it from the WebError.

Dunno if I'm clearer though

Reply:
It Is indeed way clearer. Thanks.

We would need to add a check in the error middleware thought. Something that check that, if the error catches by the middleware has a `parentError`, it uses it to compose the span instead of the error catch.